core/tests/protractor_utils/AdminPage.js

```diff
@@ -272,6 +312,144 @@ var AdminPage = function() {
       expect(name).toEqual(expectedUsernamesArray[j]);
     }
   };
+
+  this.getMiscTab = async function() {
+    await miscTabButton.click();
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.uploadTopicSimilarities =
+    async function(relativePathToSimilaritiesFile) {
+      var absPath = path.resolve(__dirname, relativePathToSimilaritiesFile);
+      await waitFor.visibilityOf(chooseSimilarityFileInput,
+        'Similarity upload form is not visible');
+      await chooseSimilarityFileInput.sendKeys(absPath);
+      await waitFor.elementToBeClickable(similarityFileUploadButton,
+        'Upload button taking too long to be clickable');
+      await similarityFileUploadButton.click();
+    };
+
+  this.expectSimilaritiesToBeUploaded = async function() {
+    await waitFor.visibilityOf(statusMessage);
+    await waitFor.elementToBeClickable(similarityDownloadButton,
+      'Similarity upload failed – download similarities button not clickable.');
+    await waitFor.textToBePresentInElement(statusMessage,
+      'Topic similarities uploaded successfully.');
+    expect(await statusMessage.getText()).toEqual(
+      'Topic similarities uploaded successfully.');
+  };
```

Comment:
Could you move the `statusMessage` checks to `uploadTopicSimiarities`? They need to run immediately after upload before the confirmation message disappears, so they should be in the same function as the upload

Reply:
done!